docs/fluvio/concepts/batching.mdx

```diff
@@ -6,10 +6,87 @@ title: "Batching"
+
+Each configuration offers a balance of benefits and trade-offs:
+
+- `max_request_size`: Allows the producer to send larger records, but may increase the risk of network congestion and latency. 
+- `batch_size`: Larger value can reduce the number of requests sent to the server, but may increase latency. Smaller value can reduce latency, but may increase the number of requests.
+- `compression`: Helps reduce storage space at the cost of increased CPU usage for compressing and decompressing data. Larger batch_size values generally improve compression ratios, especially with larger payloads.
+- `linger`: A value of 0 sends records immediately, minimizing latency but potentially reducing throughput. Higher values introduce some delay but can enhance throughput by enabling more efficient batching.
+
+The ideal parameters for the `max_request_size`, `batch_size`, `linger` and `compression` depend on your application needs.
+
+# Example Scenarios
+
+Before testing, create a topic and generate a large data file:
+
+```bash
+fluvio topic create example-topic
+printf 'This is a sample line. ' | awk -v b=500000 '{while(length($0) < b) $0 = $0 $0}1' | cut -c1-500000 > large-data-file.txt
+```
+
+### Max Request Size
+
+`max_request_size` will define the maximum size of a message that can be sent by the producer. If a message exceeds this size, Fluvio will throw an error. Even if it's a message with only one record or a batch of them.
+
+```bash
+fluvio produce example-topic --max-request-size 16384 --file large-data-file.txt --raw
+```
+
+Will be displayed the following error:
+
+```bash
+the given record is larger than the max_request_size (16384 bytes).
+```
+
+### Batch Size
+
+`batch_size` will define the maximum size of a batch of records that can be sent by the producer. If a record exceeds this size, Fluvio will process the record in a new batch without the `batch_size` as limit.
+
+```bash
+fluvio produce example-topic --batch-size 16536 --file large-data-file.txt --raw
+```
+
+There will not be any errors displayed, even if the message exceeds the batch size. But the record will be processed as a new batch.
+
+### Compression
+
+These parameters only uses the uncompressed size of the records, even with some compression algorithm is enabled.
+
+`batch_size` and `max_request_size` will only use the uncompressed message size.
+
+```bash
+fluvio produce example-topic --batch-size 16536 --compression gzip --file large-data-file.txt --raw
+fluvio produce example-topic --max-request-size 16384 --compression gzip --file large-data-file.txt --raw
+```
+
+Only the second command will display an error because the uncompressed message exceeds the max request size.
+
+
+### Linger
+
+`linger` will define the time that the producer will wait before sending a batch of records. If the batch is not full, the producer will wait for the linger time to send the batch.
+
+Even with a high linger this command will send the batch immediately because the batch is full:
```

Comment:
Even with a high linger this command will send the batch immediately because the batch is full:

=> 

As linger is only relevant when the records are smaller than the batch size, in the following example, the records are sent without delay:

Reply:
Fixed!